### ISSUE TITLE
g.E().hasId() not working

### ISSUE THREAD

**Issue by FlorianHockmann** [JanusGraph/janusgraph#1314]
`g.E().hasId()` seems to be not working:

```groovy
gremlin> edgeId = g.V(4184).bothE().id().next()
==>362-380-9hx-388
gremlin> g.E().hasId(edgeId)
gremlin> g.E(edgeId) // this works as expected
==>e[362-380-9hx-388][4176-battled->4184]
```

hasId() works correctly though when it's used mid-traversal:

```groovy
gremlin> g.V(4184).bothE().hasId(edgeId)
==>e[362-380-9hx-388][4176-battled->4184]
```

This problem was [reported on janusgraph-users by Thanos Giannakis](https://groups.google.com/d/msg/janusgraph-users/czc_ik0SpkY/UvrrGN3SCgAJ).

**Comment by RaulGuo** [comment]
I tried to fix this. waiting to be merged. [pr](https://github.com/JanusGraph/janusgraph/pull/1317)

**Comment by RaulGuo** [comment]
Hi Hockmann, I think I made a mistake, should build the pr based on some other branchs other than the master? It is blocked, does that mean that I should build another pr based on other branches? This is my first time to do so, please help to give some instruction, thanks.

**Comment by porunov** [comment]
Fixed in #2849
